### ISSUE TITLE
Bump k8s to 1.26

### ISSUE THREAD

**Issue by jmrodri** [kubernetes-sigs/controller-runtime#2035]
Creating an issue to ask for the 1.26 bump. https://www.k8s.dev/resources/release/#kubernetes-126

K8S 1.26 GA date is 2002/12/06

**Comment by lengrongfu** [comment]
@jmrodri 
"K8S 1.26 GA date is 2002/12/06"=>"K8S 1.26 GA date is 2012/12/06"

**Comment by chiukapoor** [comment]
@jmrodri @lengrongfu 
> K8S 1.26 GA date is 2002/12/06

K8S 1.26 GA date is 2022/12/06

**Comment by katexochen** [comment]
1.26 is out, and controller-runtime breaks when updating k8s Go dependencies:

```
❯ go vet ./...
# sigs.k8s.io/controller-runtime/pkg/cache
../../../../go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/cache/informer_cache.go:144:9: cannot use i.Informer (variable of type "k8s.io/client-go/tools/cache".SharedIndexInformer) as type Informer in return statement:
        "k8s.io/client-go/tools/cache".SharedIndexInformer does not implement Informer (wrong type for AddEventHandler method)
                have AddEventHandler(handler "k8s.io/client-go/tools/cache".ResourceEventHandler) ("k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration, error)
                want AddEventHandler(handler "k8s.io/client-go/tools/cache".ResourceEventHandler)
../../../../go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/cache/informer_cache.go:158:9: cannot use i.Informer (variable of type "k8s.io/client-go/tools/cache".SharedIndexInformer) as type Informer in return statement:
        "k8s.io/client-go/tools/cache".SharedIndexInformer does not implement Informer (wrong type for AddEventHandler method)
                have AddEventHandler(handler "k8s.io/client-go/tools/cache".ResourceEventHandler) ("k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration, error)
                want AddEventHandler(handler "k8s.io/client-go/tools/cache".ResourceEventHandler)
```

**Comment by xor22h** [comment]
Is there a plan for new version released? as we face same issues as @katexochen

**Comment by arkodg** [comment]
@xor22h the v0.14.0 [release](https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.14.0) has included the k8s 1.26 bump
